pkg/workloadmeta/collectors/internal/containerd/container_builder.go

```diff
@@ -100,3 +111,40 @@ func extractStatus(status containerd.ProcessStatus) workloadmeta.ContainerStatus
 
 	return workloadmeta.ContainerStatusUnknown
 }
+
+// extractIP gets the IP of a container.
+//
+// The containerd client does not expose the IPs, that's why we use the helpers
+// that we have in the "system" package to extract that information from
+// "/proc".
+//
+// A current limitation is that if a container exposes multiple IPs, this
+// function just returns one of them. That means that if a container is attached
+// to multiple networks this might not work as expected.
+func extractIP(container containerd.Container, containerdClient cutil.ContainerdItf) (string, error) {
+	taskPids, err := containerdClient.TaskPids(container)
+	if err != nil {
+		return "", err
+	}
+
+	if len(taskPids) == 0 {
+		return "", errors.New("no PIDs found")
+	}
+
+	IPs, err := system.ParseProcessIPs(
+		config.Datadog.GetString("container_proc_root"),
+		int(taskPids[0].Pid), // Any PID of the container should work
```

Comment:
I'm not sure how often this is going to be retried if there's no change to the container. As a PID can be temporary, you may want to loop on PIDs if you get an error like file not found.

Reply:
Added in this commit: https://github.com/DataDog/datadog-agent/pull/11924/commits/424f2b10cb91be38be0f594f0843a254be1b70fa